src/Aspire.Hosting/Dashboard/KubernetesDataSource.cs

```diff
@@ -0,0 +1,581 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Concurrent;
+using System.Text;
+using System.Text.Json;
+using System.Text.Json.Nodes;
+using Aspire.Dashboard.Model;
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.Dcp;
+using Aspire.Hosting.Dcp.Model;
+using Aspire.Hosting.Dcp.Process;
+using Aspire.Hosting.Utils;
+using k8s;
+using Microsoft.Extensions.Logging;
+
+namespace Aspire.Hosting.Dashboard;
+
+/// <summary>
+/// Pulls data about kubernetes resources from DCP's kubernetes API. Streams updates to consumers.
+/// </summary>
+internal sealed class KubernetesDataSource
```

Comment:
Since this is a larger refactoring, maybe it is worth to reflect in class names that this code is not related to vanilla Kubernetes, but to DCP, which has its own set of well-known object kinds. So this class could become `DcpDataSource` for example.

If we don't do it, when we get to adding real Kubernetes support, this will become quite confusing I am afraid.

Reply:
Done. However this type is internal and won't ever be public, so we can change the name easily in future if needed.